### PR TITLE
fix(Search): set default and require close button label

### LINKED ISSUE #3334: AVT 3 - React Search Component, the purpose of the X button not announced by VO
## Environment
macOS Mojave version 10.14.5
Chrome Version 75.0.3770.100 (Official Build) (64-bit)
Carbon v10 - React
Voice Over

## Detailed Description
Start VoiceOver
Navigate to the search input field
Type text in the field and the X button displays
Tab to the X button
VO announces, "Button. To click this button press Control-Option-Space"

The button should have a label that describes what the button does. (i.e. clear search)

### PR DESCRIPTION
**Pull request by emyarod**
question for reviewers: is this a target for hooks refactoring due to `getDerivedStatefromProps` usage? if so, have we determined how we want to modify our tests that currently test state changes?

---

Closes #3334

This PR sets a default close button label and makes the prop required for a11y purposes

### CODE REVIEW

File: packages/react/src/components/Search/Search.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ export default class Search extends Component {
     /**
      * Specify a label to be read by screen readers on the "close" button
      */
-    closeButtonLabelText: PropTypes.string,
+    closeButtonLabelText: PropTypes.string.isRequired,
```
Comment (asudoh) [inline packages/react/src/components/Search/Search.js]: A question; Does this work with "falling back to `defaultProps`" scenario?

Comment (emyarod) [reply packages/react/src/components/Search/Search.js]: I removed the requirement on this prop

**Comment by asudoh** [comment]
BTW let's decouple the a11y fix with the hooks thing. By doing so, would this change be ready for go out from draft mode?

**Comment by emyarod** [comment]
got it @asudoh will leave the refactor for the future. yes this is ready for review now

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @emyarod!

**Comment by joshblack** [comment]
cc @dakahn how do you feel about default aria labels?

As an aside, is there an issue for removing the double `aria-label` on button and icon?

https://github.com/carbon-design-system/carbon/blob/217d4c027b1b83157a2a46c1a3039d4f09c51d24/packages/react/src/components/Search/Search.js#L174-L180

**Comment by joshblack** [comment]
> question for reviewers: is this a target for hooks refactoring due to getDerivedStatefromProps usage? if so, have we determined how we want to modify our tests that currently test state changes?

I think for sure.

When it comes to state changes, I've been doing `wrapper.setProps` for forcing the prop change. I've been following: https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops for the implementation

**Review by jnm2377 — APPROVED** [review]

**Review by dakahn — APPROVED** [review]
Great work!
Testing using VO on latest macOS in Chrome latest; button is announced as
`Clear search input, button` 👍🏽

**Comment by snidersd** [comment]
@dakahn I retested this fix and if also fixes the DAP issue #3454

**Pull request merged**